hub/core/meta/encode/base_encoder.py

```diff
@@ -0,0 +1,177 @@
+from abc import ABC
+from typing import Any, Sequence
+from hub.constants import ENCODING_DTYPE
+import numpy as np
+
+
+class Encoder(ABC):
+    last_index_index: int = -1
```

Comment:
Is this ever _not_ equal to -1? I don't understand why this isn't a constant

Reply:
And `last_index_index` is a bad name. Would `last_row_index` be better? It's hard to tell even after having read all the code.

Reply:
the `last index` is the right-most **column**. the right-most column indicates the last index that lives in that specific row. this is the basis of the encoding.